crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -2298,6 +2308,28 @@ mod tests {
         Ok(())
     }
 
+    #[test]
+    fn string_type() -> anyhow::Result<()> {
+        let mut db = setup_db();
+
+        db.write_dedented(
+            "src/a.py",
+            r#"
+            w = "Hello"
+            x = 'world'
+            y = "Guten " + 'tag'
+            z = 'bon ' + "jour"
+            "#,
+        )?;
+
+        assert_public_ty(&db, "src/a.py", "w", "Literal[Hello]");
+        assert_public_ty(&db, "src/a.py", "x", "Literal[world]");
+        assert_public_ty(&db, "src/a.py", "y", "Literal[Guten tag]");
+        assert_public_ty(&db, "src/a.py", "z", "Literal[bon jour]");
```

Comment:
Should these have quotes in them? Or is this the expected form?

Reply:
(Quotes are necessary to disambiguate `Literal["1"]`, a string literal type, from `Literal[1]`, an int literal type.)

Reply:
That latter bit was my concern, so 👍🏼 – will push a fix for that shortly!